test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/persistence/WorkbenchPersistenceIntegrationTest.java

```diff
@@ -0,0 +1,156 @@
+    private static final String CONTAINER_ALIAS = "cont-alias";
+
+    private static DeploymentScenarioBuilderFactory deploymentScenarioFactory = DeploymentScenarioBuilderFactoryLoader.getInstance();
+    private WorkbenchWithKieServerScenario workbenchWithKieServerScenario;
+
+    private WorkbenchClientProvider workbenchClientProvider;
+    private KieServerMgmtControllerClient kieControllerClient;
+    private KieServerClientProvider kieServerClientProvider;
+    private GitProvider gitProvider;
+
+    @Before
+    public void setUp() {
+        workbenchWithKieServerScenario = deploymentScenarioFactory.getWorkbenchWithKieServerScenarioBuilder().build();
+        workbenchWithKieServerScenario.deploy();
+
+        gitProvider = GitProviderFactory.getGitProvider();
+        gitProvider.createGitRepository(workbenchWithKieServerScenario.getNamespace(), ClassLoader.class.getResource("/kjars-sources").getFile());
+
+        workbenchClientProvider = new WorkbenchClientProvider(workbenchWithKieServerScenario.getWorkbenchDeployment());
+        kieServerClientProvider = new KieServerClientProvider(workbenchWithKieServerScenario.getKieServerDeployment());
+        kieControllerClient = new KieServerMgmtControllerClient(workbenchWithKieServerScenario.getWorkbenchDeployment().getUrl().toString() + "/rest/controller",
+                workbenchWithKieServerScenario.getWorkbenchDeployment().getUsername(), workbenchWithKieServerScenario.getWorkbenchDeployment().getPassword());
+    }
+
+    @After
+    public void tearDown() {
+        InstanceLogUtil.writeDeploymentLogs(workbenchWithKieServerScenario.getWorkbenchDeployment());
+        InstanceLogUtil.writeDeploymentLogs(workbenchWithKieServerScenario.getKieServerDeployment());
+        workbenchWithKieServerScenario.undeploy();
+        gitProvider.deleteGitRepository(workbenchWithKieServerScenario.getNamespace());
+    }
+
+    @Test
+    public void testWorkbenchControllerPersistence() {
+        workbenchClientProvider.createOrganizationalUnit(ORGANIZATION_UNIT_NAME, workbenchWithKieServerScenario.getWorkbenchDeployment().getUsername());
+        workbenchClientProvider.cloneRepository(ORGANIZATION_UNIT_NAME, REPOSITORY_NAME, gitProvider.getRepositoryUrl(workbenchWithKieServerScenario.getNamespace()));
+        workbenchClientProvider.deployProject(REPOSITORY_NAME, PROJECT_NAME);
+
+        KieServerInfo serverInfo = kieServerClientProvider.getKieServerClient().getServerInfo().getResult();
+        String kieServerLocation = serverInfo.getLocation();
+
+        kieControllerClient.saveContainerSpec(serverInfo.getServerId(), serverInfo.getName(), CONTAINER_ID, CONTAINER_ALIAS, PROJECT_GROUP_ID, PROJECT_NAME, PROJECT_VERSION, KieContainerStatus.STARTED);
+
+        verifyOneServerTemplateWithContainer(kieServerLocation, CONTAINER_ID);
+
+        workbenchWithKieServerScenario.getWorkbenchDeployment().scale(0);
+        workbenchWithKieServerScenario.getWorkbenchDeployment().waitForScale();
```

Comment:
That is problematic. How to distinguish that Workbench is unavailable in different cloud environments? For example OpenShift will return HTTP 503, other providers may return HTTP 404 or any other error. The cloud environment check could be implemented in deployment ready() method and used in the test.